main/main.go

```diff
@@ -0,0 +1,226 @@
+			return
+		}
+
+		k8sCVEs := filterK8sCVEs(cves)
+		log.Infof("found %d k8s cves for year %s", len(k8sCVEs), feedName)
+		allk8sCVEs = append(allk8sCVEs, k8sCVEs...)
+
+		istioCVEs := filterIstioCVEs(cves)
+		log.Infof("found %d istio cves for year %s", len(istioCVEs), feedName)
+		allistioCVEs = append(allistioCVEs, istioCVEs...)
+	}
+
+	k8sFileName := fmt.Sprintf("%s%s%s%s", filePrefix, kubernetes, cves, fileExt)
+	k8sFileFullName := fmt.Sprintf("%s%s", path, k8sFileName)
+	utils.StoreCVEsToFile(allk8sCVEs, k8sFileFullName)
+
+	istioFileName := fmt.Sprintf("%s%s%s%s", filePrefix, istio, cves, fileExt)
+	istioFileFullName := fmt.Sprintf("%s%s", path, istioFileName)
+	utils.StoreCVEsToFile(allistioCVEs, istioFileFullName)
+
+	err = utils.WriteToBucket(kubernetes+"/"+k8sFileName, k8sFileFullName)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	log.Infof("done pushing to bucket path %q", fmt.Sprintf("%s/%s/%s", os.Getenv(utils.GcloudBucket), kubernetes, k8sFileName))
+
+	err = utils.WriteToBucket(istio+"/"+istioFileName, istioFileFullName)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	log.Infof("done pushing to bucket path %q", fmt.Sprintf("%s/%s/%s", os.Getenv(utils.GcloudBucket), istio, istioFileName))
+}
+
+func filterK8sCVEs(cves *utils.NvdCVEs) []utils.NvdCVEEntry {
+	return filterCVEs(cves, "kubernetes")
+}
+
+func filterIstioCVEs(cves *utils.NvdCVEs) []utils.NvdCVEEntry {
+	return filterCVEs(cves, "istio")
+}
+
+func filterCVEs(cves *utils.NvdCVEs, project string) []utils.NvdCVEEntry {
+	var cveEntries []utils.NvdCVEEntry
+	for _, cve := range cves.Entries {
+		for _, vendorData := range cve.CVE.Affects.Vendor.VendorData {
+			for _, productData := range vendorData.Product.ProductData {
+				if productData.ProductName == project {
+					cveEntries = append(cveEntries, cve)
+				}
+			}
+		}
+	}
+	return cveEntries
+}
+
+func getDataFeeds(dataFeedHashes map[string]string, localPath string) (map[string]string, map[string]string, error) {
+	var dataFeedNames []string
+	for y := 2010; y <= time.Now().Year(); y++ {
+		dataFeedNames = append(dataFeedNames, strconv.Itoa(y))
+	}
+
+	// Get hashes for these feeds.
+	for _, dataFeedName := range dataFeedNames {
+		hash, err := getHashFromMetaURL(fmt.Sprintf(dataFeedMetaURL, dataFeedName))
+		if err != nil {
+			log.WithError(err).WithField(logDataFeedName, dataFeedName).Warning("could not get NVD data feed hash")
+			continue
+		}
+		dataFeedHashes[dataFeedName] = hash
+	}
+
+	// Create map containing the name and filename for every data feed.
+	dataFeedReaders := make(map[string]string)
+	for _, dataFeedName := range dataFeedNames {
+		fileName := filepath.Join(localPath, fmt.Sprintf("%s%s", dataFeedName, fileExt))
+		if h, ok := dataFeedHashes[dataFeedName]; ok && h == dataFeedHashes[dataFeedName] {
+			// The hash is known, the disk should contains the feed. Try to read from it.
+			if localPath != "" {
+				if f, err := os.Open(fileName); err == nil {
```

Comment:
Nothing is being done with `f`. `os.Stat()` instead?

Reply:
I think `os.Open()` is better because it checks if the file is available for reading. `os.Stat()` merely checks for existence but doesn't checks read permissions.